File: Serene/Serene.Script/Imports/ServerImports/Administration.UserPermissionRow.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Administration
{
    [Imported, Serializable, PreserveMemberCase]
    public partial class UserPermissionRow
    {
        [InlineConstant] public const string IdProperty = "UserPermissionId";
        [InlineConstant] public const string NameProperty = "PermissionKey";
        [InlineConstant] public const string LocalTextPrefix = "Administration.UserPermission";

        public Int64? UserPermissionId { get; set; }
        public Int32? UserId { get; set; }
        public String PermissionKey { get; set; }
        public Boolean? Grant { get; set; }
        public String Username { get; set; }
        public String User { get; set; }

        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string UserPermissionId = "UserPermissionId";
            [InlineConstant] public const string UserId = "UserId";
            [InlineConstant] public const string PermissionKey = "PermissionKey";
            [InlineConstant] public const string Grant = "Grant";
            [InlineConstant] public const string Username = "Username";
            [InlineConstant] public const string User = "User";
        }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Administration.UserPermissionRow.ts
namespace Serene.Administration {
    export interface UserPermissionRow {
        UserPermissionId?: number;
        UserId?: number;
        PermissionKey?: string;
        Grant?: boolean;
        Username?: string;
        User?: string;
    }

    export namespace UserPermissionRow {
        export const idProperty = 'UserPermissionId';
        export const nameProperty = 'PermissionKey';
        export const localTextPrefix = 'Administration.UserPermission';

        export namespace Fields {
            export declare const UserPermissionId: string;
            export declare const UserId: string;
            export declare const PermissionKey: string;
            export declare const Grant: string;
            export declare const Username: string;
            export declare const User: string;
        }

        ['UserPermissionId', 'UserId', 'PermissionKey', 'Grant', 'Username', 'User'].forEach(x => (<any>Fields)[x] = x);
    }
}

